### PR TITLE
C#: Use nuget.exe from the executing machine instead of always downlo…

### PR DESCRIPTION
**Pull request by tamasvajk**
…ading it

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/NugetExeWrapper.cs [hunk 1]
```diff
@@ -219,8 +235,18 @@ private bool HasNoPackageSource()
 
         private void RunMonoNugetCommand(string command, out IList<string> stdout)
         {
-            var exe = "mono";
-            var args = $"{nugetExe} {command}";
+            string exe, args;
+            if (RunWithMono)
```
Comment (michaelnebel) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/NugetExeWrapper.cs:239]: Why is this needed?

**Review by michaelnebel — APPROVED** [review]
LGTM - just one question :-)

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/NugetExeWrapper.cs (hunk 1 above)
Comment (tamasvajk) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/NugetExeWrapper.cs:239]: It can be the case that we're on a linux/mac machine, but we need to run `nuget` instead of `mono nuget.exe`. In these cases `mono nuget` would simply fail.

For example, on my machine:
```
❯ mono /Library/Frameworks/Mono.framework/Versions/Current/Commands/nuget
Cannot open assembly '/Library/Frameworks/Mono.framework/Versions/Current/Commands/nuget': File does not contain a valid CIL image.
```

because

```
❯ cat /Library/Frameworks/Mono.framework/Versions/Current/Commands/nuget
#!/bin/sh
exec /Library/Frameworks/Mono.framework/Versions/6.12.0/bin/mono $MONO_OPTIONS /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/nuget/nuget.exe "$@"
```
